### ISSUE TITLE
createCustomElement assumes node is attached to DOM before attributes are set.

### ISSUE THREAD

**Issue by 4oo** [angular/angular#32122]
# 🐞 bug report

### Affected Package
Angular elements

### Is this a regression?

Not to my knowledge

### Description

Pull request #23825 [link](https://github.com/angular/angular/pull/23825) includes a comment that reads:
"Note that some polyfills (e.g. document-register-element) do not call the constructor. Do not assume this strategy has been created."

It then proceeds to add checks that the strategy is created in each of the callbacks.

It does not, however add the checks to the attribute [setters or getters](https://github.com/angular/angular/pull/23825/commits/a417ae43e39e87f64f8b7fe75f62c48508fd14c8#diff-ad5cb9ab336c13dbe9a23278e0299e62R186-R187).

I believe this is causing my app to explode, throwing:
```
this.ngElementStrategy is undefined
Stack trace:
set@........elements/src/create-custom-element.ts
```

## 🔬 Minimal Reproduction

Please let me know if you really need this.

I'm using the document-register-element polyfill and constructing my custom elements like this:
```ts
const node = document.createElement(SELECTOR);
node.attributeOne = "value one";
containerNode.appendChild(domNode);
```

Though I've placed the attribute assignment and the addition to the DOM next to each other, they're further separated in the application, and it would not be trivial to switch the order in which these operations are occurring.

## 🔥 Exception or Error
```
this.ngElementStrategy is undefined
Stack trace:
set@........elements/src/create-custom-element.ts
```

**Comment by 4oo** [comment]
Here's the silly workaround I'm using:

```
const node = document.createElement(SELECTOR);
node.constructor(); // see https://github.com/angular/angular/issues/32122
node.attributeOne = "value one";
containerNode.appendChild(domNode);
```

**Comment by gkalpak** [comment]
This has been fixed in #36114 (and #37226 for 9.1.x). The fix is in the latest v9 (9.1.9) and v10-RC.
